internal/storage/image.go

```diff
@@ -661,36 +877,48 @@ func (svc *imageService) PullImage(ctx context.Context, imageName RegistryImageR
 // NOTE: That means this code can run in a separate process, and it should not access any CRI-O global state.
 //
 // It returns a c/storage ImageReference for the destination.
-func pullImageImplementation(ctx context.Context, lookup *imageLookupService, store storage.Store, imageName RegistryImageReference, options *ImageCopyOptions) (types.ImageReference, error) {
+func pullImageImplementation(ctx context.Context, lookup *imageLookupService, store storage.Store, imageName RegistryImageReference, options *ImageCopyOptions) (types.ImageReference, string, error) {
 	srcSystemContext, srcRef, err := lookup.prepareReference(options.SourceCtx, imageName)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	destRef, err := istorage.Transport.NewStoreReference(store, imageName.Raw(), "")
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	policy, err := signature.DefaultPolicy(options.SourceCtx)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	policyContext, err := signature.NewPolicyContext(policy)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
-	_, err = copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
+	manifestBytes, err := copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
 		SourceCtx:        srcSystemContext,
 		DestinationCtx:   options.DestinationCtx,
 		OciDecryptConfig: options.OciDecryptConfig,
 		ProgressInterval: options.ProgressInterval,
 		Progress:         options.Progress,
 	})
 	if err != nil {
-		return nil, err
+		return nil, "", err
+	}
+
+	imageParsed, err := reference.ParseNormalizedNamed(imageName.Raw().String())
```

Comment:
??! Why is this round-tripping through text?

Reply:
I am looking for a way to convert `references.RegistryImageReference` to `reference.Named`. Any clue?

Reply:
See in the other recent thread.